shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/RouteCardData.kt

```diff
@@ -85,28 +106,27 @@ data class RouteCardData(private val lineOrRoute: LineOrRoute, val stopData: Lis
             // annoying
             if (globalData == null) return null
 
+            val hideNonTypicalPatternsBeyondNext: Duration? =
+                when (context) {
+                    Context.NearbyTransit -> 120.minutes
+                    Context.StopDetailsFiltered -> 120.minutes
+                    Context.StopDetailsUnfiltered -> null
+                }
+
             val cutoffTime = hideNonTypicalPatternsBeyondNext?.let { filterAtTime + it }
 
             return ListBuilder()
                 .addStaticStopsData(stopIds, globalData)
-                .addUpcomingTrips(schedules, predictions, filterAtTime)
-                .filterIrrelevantData(
-                    cutoffTime,
-                    showAllPatternsWhileLoading = context == Context.StopDetails,
-                    filterCancellations = context == Context.NearbyTransit
-                )
-                .addAlerts(
-                    alerts,
-                    includeMinorAlerts = context == Context.StopDetails,
-                    filterAtTime
-                )
+                .addUpcomingTrips(schedules, predictions, filterAtTime, globalData)
+                .filterIrrelevantData(filterAtTime, cutoffTime, context, globalData)
+                .addAlerts(alerts, includeMinorAlerts = context.isStopDetails(), filterAtTime)
                 .build()
                 .sort(sortByDistanceFrom, pinnedRoutes)
         }
     }
 
     class ListBuilder() {
-        var data: ByLineOrRouteBuilder = emptyMap()
+        var data: ByLineOrRouteBuilder = mutableMapOf()
```

Comment:
My original implementation plan called out making code functional where possible to help with readability. I think the use of mutability here is limited enough while building intermediate state, and hopefully the other readability improvements made throughout these classes makes things generally easier to follow. If there are pieces of the mutations that are hard to follow though, we can change approach.